### PR TITLE
Dashboard Updates and UI Patches

### PR DESCRIPTION
**Pull request by shivanmodha**
## UI Patches
- Added build environment to disable `sourcemap` files
- Added mobile height threshold #152
- Disabled flying logos for #150 until such time the code can be migrated to anime.js (controlled through `defaults.js`)
- Updated button hover/active colors with css variables:
```css
--hover-color: rgba(255, 255, 255, 0.5);
--hover-background-color: #26E8BDA0;
--active-color: rgba(255, 255, 255, 0.5);
--active-background-color: #5A7A96A0;
```
- New favicon!

## Dashboard Updates
- Modularized `UserProfileForm`
- Multiple views, `Edit` and `Overview` #151
- Checkboxes linked to `unregistered` status #151
- Updated form validation #142

## Images
image
image
image
image

### CODE REVIEW

**Comment by ryanisaacg** [comment]
@shivanmodha So from what I see, the code looks fine, but the "view" mode for the form looks weird, because of the amount of whitespace.

File: src/components/Dashboard/Forms/UserProfileForm/CustomAVInput.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * AVField wrapper object that allows for a custom input for validation
+ * Expects props: {
+ *     name: String,
+ *     label: React Renderable,
+ *     value: String,
+ *     validate: JSON Object
+ * }
+ */
+class CustomAVInput extends Component {s
```
Comment (ryanisaacg) [inline src/components/Dashboard/Forms/UserProfileForm/CustomAVInput.jsx]: Does this get used anywhere?

File: src/components/Dashboard/Forms/UserProfileForm/UserProfileForm.jsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                    </AvCheckboxGroup>
+                    <div style={{ width: "100%" }} align="right">
+                        <Button style={{ backgroundColor: theme.accent[0], marginRight: 10 }} type="reset" >Clear</Button>
+                        <Button style={{ backgroundColor: theme.primary[0] }} type="submit" >Update</Button>
+                    </div>
+                </AvForm>
+            )
+        } else {
+            let pStyle = {
+                color: theme.disabled[0] + "70", padding: 5, height: 35
```
Comment (thekabistro) [inline src/components/Dashboard/Forms/UserProfileForm/UserProfileForm.jsx]: Why `+70` on the color?

Comment (thekabistro) [reply src/components/Dashboard/Forms/UserProfileForm/UserProfileForm.jsx]: I don't like the purple at all.  Let's stick with the white?

File: src/components/Dashboard/ProfileMessage.jsx [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
+ * Renders an alert based on the message
+ * @param {String} message The displayed message 
+ */
 const ProfileMessage = ({ message }) => (
-    message && <UncontrolledAlert
-        color={message.success ? 'success' : 'danger'}
-        style={{ background: "rgba(0, 255, 0, 0.25)", border: "none", color: "white" }}
-    >
+    message && <UncontrolledAlert color={message.success ? 'success' : 'danger'} style={{ background: "rgba(0, 255, 0, 0.25)", border: "none", color: "white" }}>
         { message.value }
```
Comment (thekabistro) [inline src/components/Dashboard/ProfileMessage.jsx]: For the sake of readability, could this be reverted?

File: src/components/Profile.js [hunk 4]
```diff
@@ -278,7 +277,6 @@ class Profile {
                 },
                 json: true
             }, (error, response, body) => {
-                console.log(body);
```
Comment (thekabistro) [inline src/components/Profile.js]: Is there a reason to remove these?  I'm thinking we should have these console printouts everywhere there is an error as that'll help whenever we need to debug.

File: src/components/Dashboard/Forms/UserProfileForm/UserProfileForm.jsx (hunk 2 above)
Comment (shivanmodha) [reply src/components/Dashboard/Forms/UserProfileForm/UserProfileForm.jsx]: Give it an opacity

File: src/components/Profile.js (hunk 4 above)
Comment (shivanmodha) [reply src/components/Profile.js]: I didn't think it wise to data dump user info in the console like that

File: src/components/Dashboard/Forms/UserProfileForm/CustomAVInput.jsx (hunk 1 above)
Comment (shivanmodha) [reply src/components/Dashboard/Forms/UserProfileForm/CustomAVInput.jsx]: Alot

**Comment by shivanmodha** [comment]
@ryanisaacg noted for future pr's, but it'd be annoying and counter productive to revert the code for this pr for the sake of separation... Also my goals for this pr was more than just check boxes. It was to revamp the entire dashboard and address the issues brought up web-issues, which is why I included the check boxes, form validation, modularization and tiny 1 liner fixes here and there in one pr.

File: src/components/Profile.js (hunk 4 above)
Comment (thekabistro) [reply src/components/Profile.js]: @ryanisaacg thoughts? 

I think I'd rather have the information available in the console to debug from.  The only they'll see if about their user which they have access to anyways through the dashboard and have explicitly told us.

Comment (shivanmodha) [reply src/components/Profile.js]: I'll add it back in, but if this is the intention, we should create a [standard logger](https://github.com/pimterry/loglevel) for the application. I'm pretty sure I added those console logs in there willy nilly while debugging those magiclinks, and then forgot to remove them before the commit.

Comment (shivanmodha) [reply src/components/Profile.js]: ^ that logger is very pythonic

Comment (ryanisaacg) [reply src/components/Profile.js]: There's nothing sensitive about the log, because any info is from the user.

**Comment by ryanisaacg** [comment]
@shivanmodha 👍

**Pull request merged**